### ISSUE TITLE
MongoDB - Batch Size On Aggregate Options

### ISSUE THREAD

**Issue by RronJahiri** [quarkusio/quarkus#11565]
**Describe the extension**
On AggregateOptions in **io.quarkus.mongodb** package add batchSize variable because we want option that can get documents with batches not all documents at once.

Interested in this extension, please +1 via the emoji/reaction feature of GitHub (top right).

**Configuration suggestion**
```properties
 public <T> AggregatePublisher<T> apply(AggregatePublisher<T> stream) {
        AggregatePublisher<T> publisher = stream;
/*before code
        if (hint != null) {
            publisher = publisher.hint(hint);
        }
        if (comment != null) {
            publisher = publisher.comment(comment);
        }
        if (collation != null) {
            publisher = publisher.collation(collation);
        }
        publisher.bypassDocumentValidation(bypassDocumentValidation);
        publisher.allowDiskUse(allowDiskUse);
        if (maxAwaitTime > 0) {
            publisher.maxAwaitTime(maxAwaitTime, maxAwaitTimeUnit);
        }
        if (maxTime > 0) {
            publisher.maxAwaitTime(maxTime, maxTimeUnit);
        }
*/
/* added part
        if(batchSize > 0){
            publisher.batchSize(batchSize);
        }
*/
        return publisher;
    }

```

**Additional context**
(Add any other context about the proposal here.)

**Comment by loicmathieu** [comment]
@RronJahiri this should be easily feasible ;)

I can do it or you can provide a PR if you want to contribute ?

**Comment by RronJahiri** [comment]
> or you can provide a PR if you want to

Okay i will do a PR. 

Thanks !

**Comment by RronJahiri** [comment]
@loicmathieu sorry for this late response! I cannot do a PR because of some issues when getting Quarkus project. Could you please implement it? 

The same should also be implemented in the following class - `io.quarkus.mongodb.FindOptions`

Thank you.

**Comment by loicmathieu** [comment]
OK, I add this to my TODO list.

**Comment by lucasscharf** [comment]
May I try handle this issue?

**Comment by loicmathieu** [comment]
@lucasscharf someone already provides a pr for it.

**Comment by loicmathieu** [comment]
Fixed by #12489
